Readability.js

```diff
@@ -978,16 +987,16 @@ Readability.prototype = {
    * @param Element
    * @return number (float)
   **/
-  _getLinkDensity: function(e) {
-    var links = e.getElementsByTagName("a");
-    var textLength = this._getInnerText(e).length;
+  _getLinkDensity: function(element) {
+    var textLength = this._getInnerText(element).length;
     var linkLength = 0;
 
-    for (var i = 0, il = links.length; i < il; i += 1) {
-      linkLength += this._getInnerText(links[i]).length;
-    }
+    // XXX implement _reduceNodeList?
+    this._forEachNode(element.getElementsByTagName("a"), function(linkNode) {
+      linkLength += this._getInnerText(linkNode).length;
+    });
 
-    return linkLength / textLength;
+    return textLength !== 0 ? linkLength / textLength : 0;
```

Comment:
I see. In that case, can we just return early before calling getElementsByTagName if `textLength === 0`? If that is the case, clearly `linkLength` is also going to be 0 anyway...